src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -577,9 +622,31 @@ private IVariableInformation[] ExpandVisualized(IVariableInformation variable)
                     {
                         continue;
                     }
+
                     uint totalSize = 0;
-                    string val = GetExpressionValue(item.Size, variable, visualizer.ScopedNames);
-                    totalSize = MICore.Debugger.ParseUint(val, throwOnError: true);
+                    int rank = 0;
+                    uint[] dimensions = null;
+
+                    if (!string.IsNullOrEmpty(item.Rank))
+                    {
+                        totalSize = 1;
+                        rank = Int32.Parse(item.Rank, CultureInfo.InvariantCulture);
```

Comment:
>item.Rank [](http://example.com/codeflow?start=43&length=9)

Rank doesn't need to be a number. It can be any C++ expression. This should probably do a `TryParse, and if it can't be parsed, evaluate the expression.